src/main/kotlin/cacophony/semantic/ASTGeneration.kt

```diff
@@ -292,11 +292,48 @@ private fun generateASTInternal(
     return Empty(range = Pair(Location(0), Location(0)))
 }
 
+private fun wrapInFunction(originalAST: AST): AST {
+    val beforeStart = Location(-1)
+    val behindEnd = Location(originalAST.range.second.value + 1)
+    val program =
+        Definition.FunctionDeclaration(
+            Pair(beforeStart, behindEnd),
+            "<program>",
+            Type.Functional(
+                Pair(beforeStart, beforeStart),
+                emptyList(),
+                Type.Basic(Pair(beforeStart, beforeStart), "Unit"),
+            ),
+            emptyList(),
+            Type.Basic(Pair(beforeStart, beforeStart), "Unit"),
+            Block(
+                Pair(Location(0), behindEnd),
+                listOf(
+                    originalAST,
+                    Empty(Pair(behindEnd, behindEnd)),
+                ),
+            ),
+        )
+    val programCall =
+        FunctionCall(
+            Pair(behindEnd, behindEnd),
+            VariableUse(Pair(behindEnd, behindEnd), "<program>"),
+            emptyList(),
+        )
+    return Block(
+        Pair(beforeStart, behindEnd),
+        listOf(
+            program,
+            programCall,
+        ),
+    )
+}
+
 fun generateAST(
     parseTree: ParseTree<CacophonyGrammarSymbol>,
     diagnostics: Diagnostics,
 ): AST {
     val prunedTree = pruneParseTree(parseTree, diagnostics)!!
-    val ast = generateASTInternal(prunedTree, diagnostics) as AST
-    return ast
+    val ast = generateASTInternal(prunedTree, diagnostics)
+    return wrapInFunction(ast)
```

Comment:
Now that the whole program is wrapped in a main function, toplevel `return ()` is valid. Could you please add a minimalistic test to document this behavior too? (Sorry if there is one already, I must have missed it)